src/templates/console-v2.html

```diff
@@ -233,6 +239,94 @@
           refreshLine();
         }
 
+        function searchHistory(query, startIndex = -1) {
+          // Search backwards through history from startIndex
+          if (startIndex === -1) {
+            startIndex = history.length - 1;
+          }
+          for (let i = startIndex; i >= 0; i--) {
+            if (history[i].includes(query)) {
+              return i;
+            }
+          }
+          return -1;
+        }
+
+        // reverse-search
+        function enterReverseSearchMode() {
+          reverseSearchMode = true;
+          reverseSearchQuery = "";
+          reverseSearchIndex = -1;
+          savedBuffer = buffer;
+          buffer = "";
+          cursorIndex = 0;
+          updateReverseSearchDisplay();
+        }
+
+        function exitReverseSearchMode(accept = true) {
+          reverseSearchMode = false;
+          if (!accept) {
+            buffer = savedBuffer;
+            cursorIndex = buffer.length;
+          }
+          reverseSearchQuery = "";
+          reverseSearchIndex = -1;
+          savedBuffer = "";
+          refreshLine();
+        }
+
+        function updateReverseSearchDisplay() {
+          const searchPrompt = `(reverse-i-search)\`${reverseSearchQuery}': `;
+          let displayText = "";
+          if (reverseSearchIndex >= 0) {
+            displayText = history[reverseSearchIndex];
+          }
+          // Clear line and write search prompt with matched command
+          term.write(`\r\x1b[K${searchPrompt}${displayText}`);
+        }
+
+        function reverseSearchAddChar(char) {
+          reverseSearchQuery += char;
+          reverseSearchIndex = searchHistory(reverseSearchQuery);
+          if (reverseSearchIndex >= 0) {
+            buffer = history[reverseSearchIndex];
+            cursorIndex = buffer.length;
+          }
+          updateReverseSearchDisplay();
+        }
+
+        function reverseSearchBackspace() {
+          if (reverseSearchQuery.length > 0) {
+            reverseSearchQuery = reverseSearchQuery.slice(0, -1);
+            if (reverseSearchQuery) {
+              reverseSearchIndex = searchHistory(reverseSearchQuery);
+              if (reverseSearchIndex >= 0) {
+                buffer = history[reverseSearchIndex];
+                cursorIndex = buffer.length;
+              }
+            } else {
+              reverseSearchIndex = -1;
+              buffer = "";
+              cursorIndex = 0;
+            }
+            updateReverseSearchDisplay();
+          }
+        }
+
+        function reverseSearchNext() {
+          if (reverseSearchQuery && reverseSearchIndex > 0) {
+            reverseSearchIndex = searchHistory(
+              reverseSearchQuery,
+              reverseSearchIndex - 1
+            );
+            if (reverseSearchIndex >= 0) {
+              buffer = history[reverseSearchIndex];
+              cursorIndex = buffer.length;
+            }
```

Comment:
This if statement appears several time so I think we can extract it to a function.